### PR TITLE
Nav kalman filter cleanup

### PR DESCRIPTION
**Pull request by maxguy2001**
## Description
clean up `kalman_filter.cpp` and `kalman_filter.hpp` in navigation

## Changes
- Remove `using` to make typing clearer
- Changed increments to prefix notation
- Fixed naming to match style guide
- Add `const` for readability

### CODE REVIEW

File: src/navigation/kalman_filter.cpp [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
+  Eigen::MatrixXf A = createStateTransitionMatrix(0.0);
+  Eigen::MatrixXf Q = createStateTransitionCovarianceMatrix();
+  Eigen::MatrixXf H = createMeasurementMatrix();
 
   // check system navigation run for R setup
-  System &sys = System::getSystem();
-  MatrixXf R  = MatrixXf::Zero(m_, m_);
+  const auto &sys   = utils::System::getSystem();
+  Eigen::MatrixXf R = Eigen::MatrixXf::Zero(m_, m_);
   ;
```
Comment (MiltFra) [inline src/navigation/kalman_filter.cpp:32]: lonely `;`

File: src/navigation/kalman_filter.cpp [hunk 2]
```diff
@@ -38,123 +40,110 @@ void KalmanFilter::setup()
...
 
   // setup initial estimates
-  VectorXf x = VectorXf::Zero(n_);
-  MatrixXf P = createInitialErrorCovarianceMatrix();
+  Eigen::VectorXf x = Eigen::VectorXf::Zero(n_);
+  Eigen::MatrixXf P = createInitialErrorCovarianceMatrix();
   kalmanFilter_.setInitial(x, P);
 }
 
 void KalmanFilter::updateStateTransitionMatrix(double dt)
```
Comment (MiltFra) [inline src/navigation/kalman_filter.cpp:48]: `const`

File: src/navigation/kalman_filter.cpp [hunk 3]
```diff
@@ -38,123 +40,110 @@ void KalmanFilter::setup()
...
 {
-  MatrixXf A = createStateTransitionMatrix(dt);
+  Eigen::MatrixXf A = createStateTransitionMatrix(dt);
   kalmanFilter_.updateA(A);
 }
 
 void KalmanFilter::updateMeasurementCovarianceMatrix(double var)
```
Comment (MiltFra) [inline src/navigation/kalman_filter.cpp:54]: `const`

Comment (MiltFra) [reply src/navigation/kalman_filter.cpp:54]: also `data::nav_t`? why is this `double`?

File: src/navigation/kalman_filter.cpp [hunk 4]
```diff
@@ -38,123 +40,110 @@ void KalmanFilter::setup()
...
 {
-  MatrixXf R = MatrixXf::Constant(m_, m_, var);
+  Eigen::MatrixXf R = Eigen::MatrixXf::Constant(m_, m_, var);
   kalmanFilter_.updateR(R);
 }
 
-const nav_t KalmanFilter::filter(nav_t z)
+const data::nav_t KalmanFilter::filter(data::nav_t z)
```
Comment (MiltFra) [inline src/navigation/kalman_filter.cpp:60]: remove leading `const`, add `const` for `z`

**Review by MiltFra — CHANGES_REQUESTED** [review]

File: src/navigation/kalman_filter.cpp [hunk 5]
```diff
@@ -45,19 +45,19 @@ void KalmanFilter::setup()
   kalmanFilter_.setInitial(x, P);
 }
 
-void KalmanFilter::updateStateTransitionMatrix(double dt)
+void KalmanFilter::updateStateTransitionMatrix(const double dt)
```
Comment (MiltFra) [inline src/navigation/kalman_filter.cpp:48]: `double`

**Review by MiltFra — APPROVED** [review]

**Review by callumTM — APPROVED** [review]
Looks all good to me!

**Pull request merged**